### PR TITLE
Add CI to ElectraWifi

### PR DESCRIPTION
**Pull request by guysoft**
So scrabbling to compile today ElectraWifi, and the unpleasant conversation in #4 made me realize that Ci/CD might make things easier and more accessible.

So I give you a github action that builds both ``firmware-esp01.bin`` and ``firmware-esp12e.bin``.

I could not compile the code without the commits in #8 (thanks @garci66 ). So I merged that in too.
Did not test the artifact yet, I will flash and update.

### CODE REVIEW

**Comment by guysoft** [comment]
Tested with ``firmware-esp12e.bin``
Auto, Cool and Fan work for me, the device identifies as homie version ``3.0.1``.

File: platformio.ini [hunk 1]
```diff
@@ -8,12 +8,30 @@
...
+;you might need to modify some code in the Homie library since the way the URL client has been handled has changed. 
+[env:esp01]
+;platform = espressif8266@2.5.0
+;target define  ARDUINO_ESP8266_ESP01
+platform = espressif8266
+board = esp01_1m
+upload_protocol = esptool
+framework = arduino
+; ADD -D ELECTRAWIFI_NO_IR_RCV  to remove the IR reception code
+build_flags = -D PIO_FRAMEWORK_ARDUINO_LWIP2_LOW_MEMORY  -D ELECTRAWIFI_NO_IR_RCV
```
Comment (giladger) [inline platformio.ini]:
```suggestion
build_flags = -D PIO_FRAMEWORK_ARDUINO_LWIP2_LOW_MEMORY 
```

**Comment by garci66** [comment]
very nice!!!

im still stuggling with the horizontal louvers. The code matches what the remote sends.. but somehow it doesnt work properly

also.. did you guys manage to get OTA working at all?

finally, in my case, it seems I had to add a power on delay to the ESP-01 (via an R-C circuit on the "enable" pin of the ESP01). Otherwise, my AC would stop responding to all Remote commands (both through homie and through the regular remote receiver as well)

but if I plugged in to the DIN port after the device was plugged in, it worked well. I thin the RC should solve it. Hopefully!!

(sorry for the off-topic comments)

finally.. have you managed any OTA updates? it seems as there wouldnt be enough flash for OTA, at least for devices with 1M

**Comment by guysoft** [comment]
@garci66 I managed to get OTA to register the device, but I didn't manage to preform an upgrade.

I used this docker container and put it up in the network using a raspberrypi: https://hub.docker.com/r/ivdata/homie-ota
Screenshot_20220310_183342

**Comment by garci66** [comment]
Yeah.. ota configuration works. But the firmware upgrade got me nowhere.

In my device, given I have 1 MB of flash, I don't think i'll be able to
upgrade unless somehow the firmware is less than half the size (500KB or
so).

for tasmota, there is a "minimal"  version thats around 300KB that gets
flashed temporarily... so for example, if your FW is 600KB, the OTA
procedure flashes the minimal image of 300KB which then flashes the new
full version of 600KB

but i have not manage to make a "small" build with homie, even with all
knobs disabled.

**Pull request merged**